app/executor.php

```diff
@@ -113,23 +113,23 @@ function logError(Throwable $error, string $action, Utopia\Route $route = null)
 };
 
 function getStorageDevice($root): Device {
-    switch (App::getEnv('_APP_STORAGE_DEVICE', Storage::DEVICE_LOCAL)) {
-        case Storage::DEVICE_LOCAL:default:
-            return new Local($root);
+    $device = App::getEnv('_APP_STORAGE_DEVICE', Storage::DEVICE_LOCAL);
+    switch ($device) {
         case Storage::DEVICE_S3:
-            $s3AccessKey = App::getEnv('_APP_STORAGE_S3_ACCESS_KEY', '');
-            $s3SecretKey = App::getEnv('_APP_STORAGE_S3_SECRET', '');
-            $s3Region = App::getEnv('_APP_STORAGE_S3_REGION', '');
-            $s3Bucket = App::getEnv('_APP_STORAGE_S3_BUCKET', '');
-            $s3Acl = 'private';
-            return new S3($root, $s3AccessKey, $s3SecretKey, $s3Bucket, $s3Region, $s3Acl);
         case Storage::DEVICE_DO_SPACES:
-            $doSpacesAccessKey = App::getEnv('_APP_STORAGE_DO_SPACES_ACCESS_KEY', '');
-            $doSpacesSecretKey = App::getEnv('_APP_STORAGE_DO_SPACES_SECRET', '');
-            $doSpacesRegion = App::getEnv('_APP_STORAGE_DO_SPACES_REGION', '');
-            $doSpacesBucket = App::getEnv('_APP_STORAGE_DO_SPACES_BUCKET', '');
-            $doSpacesAcl = 'private';
-            return new DOSpaces($root, $doSpacesAccessKey, $doSpacesSecretKey, $doSpacesBucket, $doSpacesRegion, $doSpacesAcl);
+        case Storage::DEVICE_BACKBLAZE:
+        case Storage::DEVICE_LINODE:
+        case Storage::DEVICE_WASABI:
+            $accessKey = App::getEnv('_APP_STORAGE_S3_ACCESS_KEY', '');
+            $secretKey = App::getEnv('_APP_STORAGE_S3_SECRET', '');
+            $region = App::getEnv('_APP_STORAGE_S3_REGION', '');
+            $bucket = App::getEnv('_APP_STORAGE_S3_BUCKET', '');
+            $acl = 'private';
+            /**@var $adapter Utopia\Storage\Device* */
+            $adapter = 'Utopia\\Storage\\Device\\' . $device;
```

Comment:
$device probably needs to be properly cased so that user can enter in .env in any case?